peachjam/views/taxonomy.py

```diff
@@ -11,32 +12,49 @@ class TaxonomyListView(TemplateView):
 
     def get(self, request, *args, **kwargs):
         context = self.get_context_data(**kwargs)
-        context["taxonomies"] = Taxonomy.dump_bulk()
+        context["taxonomies"] = Taxonomy.get_allowed_taxonomies(request.user)["tree"]
         context["taxonomy_url"] = "taxonomy_detail"
         return self.render_to_response(context)
 
 
-class TaxonomyFirstLevelView(DetailView):
+class AllowedTaxonomyMixin:
+    def dispatch(self, request, *args, **kwargs):
+        self.taxonomy = self.get_taxonomy()
+        is_ancestor_restricted = self.taxonomy.get_ancestors().values_list(
+            "restricted", flat=True
+        )
+        if self.taxonomy.restricted or any(is_ancestor_restricted):
+            if not request.user.has_perm("peachjam.view_taxonomy", self.taxonomy):
+                return HttpResponseForbidden()
```

Comment:
This response indicates "this exists, but you can't see it". From a security standpoint, there's an argument that we shouldn't even indicate to someone that the thing exists.

So I propose that we just return a 404 here.

Reply:
Actually, I think that may cause more confusion than it's worth. Let's keep it like this.